lib/wash/entry.rb

```diff
@@ -343,6 +354,50 @@ def cache_ttls(ttls = {})
       @cache_ttls = @cache_ttls.merge(ttls)
     end
 
+    # volumefs returns an entry that provides a view of the parent entry's filesystem.
+    # Use it when listing children. It can be used with `prefetch :list`.
+    #
+    # @example
+    #   class Foo
+    #     parent_of VOLUMEFS
+    #     def list
+    #       [volumefs('fs', maxdepth: 5)]
+    #     end
+    #   end
+    #
+    # @param [String] name The name to use for the child representing the remote filesystem.
+    # @param [Hash] options Options to configure the filesystem.
+    #     - maxdepth: How deep to walk the remote filesystem when needed. This is a trade-off
+    #       between speed of the transport and filesystem complexity. For fast transports it
+    #       should be low (1-3), for slower transports higher (3-6).
+    def volumefs(name, options = {})
+      { type_id: VOLUMEFS, name: name, state: options.to_json }
+    end
+
+    # transport requests that Wash use a built-in transport with the supplied
+    # transport options to implement Exec. When using this feature, you must
+    # also declare the 'exec' method so it's included in the schema. You may
+    # use this to implement a fallback implementation for entries where the
+    # transport feature is not used.
+    #
+    # @example
+    #   class Foo
+    #     def initialize(name)
+    #       transport :ssh, host: name, user: 'root'
+    #     end
+    #
+    #     def exec
+    #       raise 'implemented by transport'
+    #     end
+    #   end
+    #
+    # @param [Symbol] klass The transport to use. Only `:ssh` is supported
+    # @param [Hash] options For SSH: :host (required), :user, :password,
+    #     :fallbackuser, :identityfile, :knownhosts, :hostkeyalias, :retries
+    def transport(klass, options = {})
```

Comment:
We could define the `exec` for them and save some clutter.

Reply:
So that's challenging because the `exec` is part of the class and can be used in defining the schema, but the `transport` calls happen during instance initialization. I'd still have to have something called during class definition.

Reply:
Ahh yeah you're right, we do need a class-level defn. for entry schemas so your example's correct. Would be nice to add a comment about that somewhere because it seems like that'd be a common question.

Reply:
That's right in the code comment about 12 lines up.